src/Explorer/Controls/Settings/SettingsComponent.tsx

```diff
@@ -648,6 +667,39 @@ export class SettingsComponent extends React.Component<SettingsComponentProps, S
   private onComputedPropertiesDirtyChange = (isComputedPropertiesDirty: boolean): void =>
     this.setState({ isComputedPropertiesDirty: isComputedPropertiesDirty });
 
+  private onDataMaskingContentChange = (newDataMasking: DataModels.DataMaskingPolicy): void => {
+    if (!newDataMasking.excludedPaths) {
+      newDataMasking.excludedPaths = [];
+    }
+    if (!newDataMasking.includedPaths) {
+      newDataMasking.includedPaths = [];
+    }
+
+    const validationErrors = [];
+    if (!Array.isArray(newDataMasking.includedPaths)) {
+      validationErrors.push("includedPaths must be an array");
+    }
+    if (!Array.isArray(newDataMasking.excludedPaths)) {
+      validationErrors.push("excludedPaths must be an array");
+    }
+    if (typeof newDataMasking.policyFormatVersion !== "number") {
+      validationErrors.push("policyFormatVersion must be a number");
+    }
+    if (typeof newDataMasking.isPolicyEnabled !== "boolean") {
+      validationErrors.push("isPolicyEnabled must be a boolean");
+    }
+
+    if (validationErrors.length > 0) {
```

Comment:
If validation fails, we should display the errors somewhere. Otherwise, user doesn't know why the Save button is disabled.